core/alfred_core.py

```diff
@@ -20,22 +25,44 @@ def __init__(self, tts=None, stt=None ):
         tts (function, optional): Text to speech function. Should accept text. Defaults to None.
         stt (function, optional): Speech to text function. Should return text. Defaults to None.
     """
+    self.audioui = AudioUI()
     if tts:
       self._tts = tts
     if stt:
       self._stt = stt
     else:
       self._recognizer = sr.Recognizer() # Initialize a Speech Recognizer object
       self._mic = sr.Microphone() # initialize a microphone
-  
 
+
+  def say(self, text):
+    """Turns a string of text into audible speech
+    Output:
+        Audible speech through device speaker
+    Args:
+        text (string): text to turn into speech
+    """
+    self.audioui.say(text)
+    return
+  def listen(self):
+    """Record audio with mic and returns text
+    Output:
+        return: text (string): text from audio text
+    Input:
+        audio from device mic
+    """
+    return self.audioui.listen()
+
+
+    '''
   def say(self, text):
```

Comment:
Can you remove this? Looks like it got left when you added your new listen/say functions. Looks good though!